base/loading.jl

```diff
@@ -1218,11 +1218,9 @@ function include_package_for_output(pkg::PkgId, input::String, depot_path::Vecto
     end
 end
 
-@assert precompile(include_package_for_output, (PkgId,String,Vector{String},Vector{String},Vector{String},typeof(_concrete_dependencies),Nothing))
```

Comment:
I'm not looking at the context here (in the middle of other things), but if memory serves it's actually an error if this doesn't get precompiled?

Reply:
We still shouldn't use `@assert` here. A future minor release of Julia is allowed to disable assertions by default. If you really need to throw an error here, you should do it explicitly.

Reply:
E.g. 
```
result = precompile(include_package_for_output, (PkgId,String,Vector{String},Vector{String},Vector{String},typeof(_concrete_dependencies),Nothing))
if !result
    msg = "failed to precompile"
    throw(ErrorException(msg))
end
```

Or something like that.

That will ensure that the error is always thrown, even if assertions are disabled by default in a future release of Julia.